File: generic-method.lua
local hello = {
   print = function (self, value)
      for i=1,self.times do
         print(value)
      end
   end
}

function hello.new(times)
   return setmetatable({times = times or 1}, {__index = hello})
end

h = hello.new(2)
h:print"cool"
h:print(11)

File: generic-method.cc
#include <iostream>

class Hello {
public:
    int times;
    Hello(int times) : times(times) {}
    template <typename T>
    void print(T value) {
        for(int i=0; i<times; ++i) {
            std::cout << value << "\n";
        }
    }
};

int main() {
    Hello h {2};
    h.print("cool");
    h.print(11);
    return 0;
}
